[docs] Edits for the 'requirements' in ReadMe.md (#2035)
Haivision/srt: README.md

## Requirements

* C++03 (or above) compliant compiler.
* CMake 2.8.12 or above (as build system).
* OpenSSL 1.1 (to enable encryption, or build with `-DENABLE_ENCRYPTION=OFF`).

> * Multithreading is provided by either of the two:

Reviewer: ```suggestion
* Multithreading is provided by either of the following:
```

  * C++11: standard library (std);

>   * C++03: Pthreads (for POSIX systems it's builtin, for Windows there is a ported library).

Reviewer: ```suggestion
  * C++03: Pthreads (for POSIX systems it's built in, for Windows there is a ported library).
```

* Tcl 8.5 (optional, used by `./configure` script or use CMake directly).

> For detailed description of the build system and options, please refer to [SRT Build Options](docs/build/build-options.md).

Reviewer: ```suggestion
For a detailed description of the build system and options, please refer to [SRT Build Options](docs/build/build-options.md).
```
